### PR TITLE
Adding customer specific partner find hook

### PR DESCRIPTION
**Pull request by gfcapalbo**
Adding the partner recognition logic to the hook left for this purpose by alexis in the code. Want to merge in one branch.

### CODE REVIEW

File: base_business_document_import/models/business_document_import.py [hunk 1]
```diff
@@ -179,7 +179,24 @@ def _match_partner(
...
+        if partner_dict.get('email'):
+            email = partner_dict['email'].replace(' ', '')
+            partner = self.env['res.partner'].search(
+                domain + [('email' , '=', email)], limit=1
+            )
+            # we bypass it all with a perfect email hit.
+            if partner:
+                return partner
+        if partner_dict.get('name'):
+            partner = self.env['res.partner'].search(
```
Comment (hbrunn) [inline base_business_document_import/models/business_document_import.py]: use `name_search` here

File: base_business_document_import/models/business_document_import.py [hunk 2]
```diff
@@ -179,7 +179,24 @@ def _match_partner(
     @api.model
     def _hook_match_partner(
             self, partner_dict, chatter_msg, domain, partner_type_label):
-        return False
+        # looking for the exact contact, we use email and contact name.
+        # if the email is a perfect hit we proceed with that, if not we use the
+        # partner name contained in all partners of our company
+        partner = False
+        if partner_dict.get('email'):
+            email = partner_dict['email'].replace(' ', '')
```
Comment (hbrunn) [inline base_business_document_import/models/business_document_import.py]: do you mean `.strip()`?

**Review by hbrunn — CHANGES_REQUESTED** [review]
if this is customer specific, this shouldn't be here but in the customer module. But I don't see where this code is customer specific, so I'd rather have this here.
Please squash your commits after following up

**Comment by hbrunn** [comment]
I think the issue at hand should be solved more general anyways: Currently, the fields for `partner_id` are prepared in https://github.com/hbrunn/edi/blob/9.0-saleorder_import_invoice2data/sale_order_import_invoice2data/wizard/sale_order_import_pdf.py#L78 and used in https://github.com/hbrunn/edi/blob/9.0-saleorder_import_invoice2data/sale_order_import/wizard/sale_order_import.py#L138

You should adapt the latter such that we can pass a similar dict for `partner_invoice` and partner_shipping`, or you extend `match_partner` such that we can select a partner (the company) the way it works now, but search for a contact person of that partner with some other keys (like `contact_name`, `contact_email` etc)

File: base_business_document_import/models/business_document_import.py (hunk 2 above)
Comment (gfcapalbo) [reply base_business_document_import/models/business_document_import.py]: would be better, anyway emails never have spaces. will change

**Comment by gfcapalbo** [comment]
@hbrunn changes added. Squashed.

**Comment by gfcapalbo** [comment]
@hbrunn this branch has been tested for a while and works. commits squashed. match_partner extended as asked.

**Pull request merged**